### ISSUE TITLE
[Bug]: [NRT] Baseset road sprites (1313-1331) are drawn underneath roadtypes in cities; should be terrain sprites instead

### ISSUE THREAD

**Issue by Andrew350** [OpenTTD/OpenTTD#11154]
### Version of OpenTTD

any

### Expected result

As is consistent with the "rural" roads (those in town zone 0), "city" roads (those in town zones 1-4) should also have terrain sprites drawn underneath when a roadtype set is used, not the baseset road sprites.

### Actual result

Baseset road sprites are drawn instead of terrain sprites.

### Steps to reproduce

1) Load roadtype set
2) Use sprite picker tool
3) Notice baseset road sprites (1313-1331) are drawn underneath city roads instead of terrain sprites, unlike with rural roads

It makes no sense to have baseset road sprites drawn underneath NewGRF-supplied ones. It should be the responsibility of the NRT author to handle covering the entire road tile properly including sidewalks, rather than (as it is done now) relying on exploiting a single action 5 to replace the base sprites to avoid covering the full tile per each roadtype. It is much easier in most cases, which is why we do it, but ultimately it is relying on incorrect behavior. It also causes issues for roadtypes which are _not_ meant to cover the full tile with pavements, since in those cases we must currently use said action 5 to add our own custom terrain sprites in order to work properly, which should not be necessary.

**Comment by PeterN** [comment]
I don't think this is a bug; this is intentional behaviour. The intention is that the overlay sprite is drawn on top and leaves the pavements (mostly) intact.

It is the town's choice whether pavements are drawn, and this is not (currently) delegated to the road type.

**Comment by auge8472** [comment]
First things first: I was the one who asked if and how it is possible to replace the pavement roads (see [the corresponding forum thread](https://www.tt-forums.net/viewtopic.php?t=91072)).

> It is the town's choice whether pavements are drawn, and this is not (currently) delegated to the road type.

Generally I'm o.k. with the argument but sometimes (with (a) specific road type(s)) I as the set author wish to be able to decide if the specific road type should show pavements or not. As an example, I would like to mention footpaths or motorways/Autobahn that are not to be provided with a pavement even within cities. See the following screenshots. Please pay attention to the road tiles inside the block of buildings.

First shot: On the left a slim footpath and on the right a motorway/Autobahn, taking around 90% of the tiles width, both with their original width.

![Bildschirmfoto zu 2023-07-27 13-21-15](https://github.com/OpenTTD/OpenTTD/assets/1734660/e71de687-a0aa-4085-9a88-f7fd4f053f9b)

Second shot: The same situation a few game days later with pavement underlays for the example road types. While it is only a small graphical issue for the motorway on the right, it is a complete optical fail for the footpath on the left.

![Bildschirmfoto zu 2023-07-27 13-22-00](https://github.com/OpenTTD/OpenTTD/assets/1734660/4fc07ea8-4d57-487e-aa07-bb09d49945e0)

And at this point it is counterintuitive to have to replace the baseset sprites one by one. Futhermore it is IMHO a mess to do this for the original baseset ***and*** for the OpenGFX-set (and for possibly encountering further base sets).

A compromise would be to be able to replace one base set sprite with another one (replace pavement roads with grass, sand or snow) instead to have to provide the replacement graphics as part of the set *or* to be able to overrule the game mechanic to leave the decision to the town.

**Comment by 2TallTyler** [comment]
What if roadtypes (and tramtypes) gained a `ROADTYPE_FLAG_NO_SIDEWALKS` flag? Then the author could choose if they want a sidewalk or not, and existing NewGRFs are not affected?

**Comment by Andrew350** [comment]
That sounds like a great solution to me; the best of both worlds 🙂

**Comment by PeterN** [comment]
I already started a patch for letting roads have input into roadside flags, but... forgot about it...

**Comment by auge8472** [comment]
Came across the issue after two years again.

@PeterN The 👍 for the patch, the 😕 for forgetting it. 😄

I'll be patient.

**Comment by PeterN** [comment]
Whatever my patch was it wouldn't fix this underlay/overlay issue.

The main issue is that the baseset does not contain the bare pavement sprites necessary. There is a flat one, but nothing for sloped tiles.
And then, drawing kerbs would have to be the responsibility of the NewGRF. There's currently no variable to test the road's ground type.

**Comment by PikkaBird** [comment]
>It should be the responsibility of the NRT author to handle covering the entire road tile properly

> There's currently no variable to test the road's ground type.

I test the town zone, and draw full-tile road sprites with or without pavement as appropriate. This does create unclean graphic updates as the town zones change - which I guess is potentially a bug, if we consider this method correct (and why would we not, since the var exists?).

<img width="454" height="303" alt="Image" src="https://github.com/user-attachments/assets/9e7c0dbd-ee4d-4a2e-90e4-249a4c754015" />

**Comment by PeterN** [comment]
Eh, town zone is only loosely related to the actual ground type which is indeed, not updated immediately. The correct approach is to just add a variable to expose the current ground type, not claim there's a bug in some other variable.

**Comment by PikkaBird** [comment]
> just add a variable to expose the current ground type

That would work for me :)
